Add 0.62 blog post (#1778)
facebook/react-native-website: website/blog/2020-03-24-version-0.62.md

---
title: Announcing React Native 0.62 with Flipper
author: Rick Hanlon
authorTitle: React Native Core at Facebook
authorURL: https://twitter.com/rickhanlonii
authorImageURL: https://avatars3.githubusercontent.com/u/2440089?s=460&v=4
authorTwitter: rickhanlonii
category: announcements
---

Today we’re releasing React Native version 0.62 which includes support for Flipper by default.

> This release comes in the midst of a global pandemic. We’re releasing this version today to respect the work of hundreds of contributors who made this release possible and to prevent the release from falling too far behind master. Please be mindful of the reduced capacity of contributors to help with issues and prepare to delay upgrading if necessary.

Reviewer: 👍

> This release comes in the midst of a global pandemic. We’re releasing this version today to respect the work of hundreds of contributors who made this release possible and to prevent the release from falling too far behind master. Please be mindful of the reduced capacity of contributors to help with issues and prepare to delay upgrading if necessary.

Reviewer 2: ```suggestion
This release comes in the midst of a global pandemic. We’re releasing this version regardless to respect the work of hundreds of contributors who made this release possible and to prevent the release from falling too far behind master. Please be mindful of the reduced capacity of contributors to help with issues and prepare to delay upgrading if necessary.
```

Making it a bit less ephemeral. Don’t feel strong about it, though.

Author: My goal here _is_ to be ephemeral - readers in the future after the crisis should understand that the reference point is the day of release. The issue with "regardless" or "nevertheless" is that they imply "in spite of" but we're saying here that the release _is_ being affected by the crisis.

Reviewer 2: Gotcha, thanks 👌

## Flipper by default

[Flipper](https://fbflipper.com/) is a developer tool for debugging mobile apps. It’s popular in the Android and iOS communities, and in this release we’ve enabled support by default for new and existing React Native apps.

Flipper provides the following features out of the box:

- Network Inspector

> - Logout Inspector

Reviewer 2: Is this supposed to be ‘layout’? Or did I miss another plugin? 🙀 

```suggestion
- Layout Inspector
```

- Crash Reporter
- Device logs

> There is also a marketplace of custom Flipper plugins you can install:

Reviewer 2: Wondering if this should be slightly expanded to make it clear anybody can ‘easily’ make these?

Author: Yeah, was hoping someone could help expand here cc @passy @mweststrate

[TODO: Flipper Image]

> For more information, check out the Flipper documentation here.

Reviewer 2: ```suggestion
For more information, check out the Flipper documentation [here](https://fbflipper.com/docs/getting-started.html).
```

Author: The correct link is below, will update to use that 👍

> For more information, check out the Flipper documentation here.

Reviewer 3: The correct URL will be: https://fbflipper.com/docs/features/react-native.html

It is not yet online, but hopefully later today. I'll update here if it is online.

> For more information, check out the Flipper documentation here.

Author: ```suggestion
For more information, check out the [Flipper documentation](https://fbflipper.com/docs/features/react-native.html).
```

## Better dark mode support

> We’ve added a new Appearance module to provide access to the user's appearance preferences, such as their preferred color scheme (light or dark).

Reviewer 2: ```suggestion
We’ve added a new `Appearance` module to provide access to the user's appearance preferences, such as their preferred color scheme (light or dark).
```

```
const colorScheme = Appearance.getColorScheme();
if (colorScheme === 'dark') {
  // Use dark color scheme
}
```

> We’ve also added a hook to subscribe to state updates to the users preferences automatically:

Reviewer 2: Unsure, but didn’t read as smooth to my non-native brain.

```suggestion
We’ve also added a hook to subscribe to state updates to the users preferences:
```

Author: Makes sense, subscriptions are automatic updates

```
import {Text, useColorScheme} from 'react-native';

const MyComponent = () => {
  const colorScheme = useColorScheme();
  return <Text>useColorScheme(): {colorScheme}</Text>;
};
```

See the docs for [Appearance](https://reactnative.dev/docs/appearance) and [useColorScheme](https://reactnative.dev/docs/usecolorscheme) for more information.

> ## Moving AppleTV to react-native-tvos

Reviewer 2: ```suggestion
## Moving Apple TV to react-native-tvos
```

> As part of our Lean Core effort and to bring AppleTV in line with other platforms like React Native Windows and React Native MacOS, we’ve started to remove AppleTV specific code from core.

Reviewer: ```suggestion
As part of our [Lean Core effort](/LINK) and to bring AppleTV in line with other platforms like React Native Windows and React Native MacOS, we’ve started to remove AppleTV specific code from core.
```
nit: maybe worth adding the link to the lean core issue? (is that still open? it's been quite a long time since last I've checked)

Reviewer 4: Looks like it's still open https://github.com/facebook/react-native/issues/23313

```suggestion
As part of our [Lean Core effort](https://github.com/facebook/react-native/issues/23313) and to bring AppleTV in line with other platforms like React Native Windows and React Native MacOS, we’ve started to remove AppleTV specific code from core.
```

Reviewer 2: There’s also https://reactnative.dev/blog/#lean-core

Author: ```suggestion
As part of our [Lean Core effort](https://reactnative.dev/blog/#lean-core) and to bring Apple TV in line with other platforms like React Native Windows and React Native MacOS, we’ve started to remove Apple TV specific code from core.
```

> As part of our Lean Core effort and to bring AppleTV in line with other platforms like React Native Windows and React Native MacOS, we’ve started to remove AppleTV specific code from core.

Reviewer 2: (Double-checking the `macOS` part with the team.)

```suggestion
As part of our Lean Core effort and to bring Apple TV in line with other platforms like React Native Windows and React Native macOS, we’ve started to remove Apple TV specific code from core.
```

Reviewer 2: Yup, ‘React Native macOS’ it is 👍

Going forward, Apple TV support for React Native will be maintained in [react-native-community/react-native-tvos](https://github.com/react-native-community/react-native-tvos) along with the corresponding `react-native-tvos` NPM package. This is a full fork of the main repository, with only the changes needed to support Apple TV.

> Releases of `react-native-tvos` will be based on the public release of React Native. For this 0.62 release of `react-native` please upgrade AppleTV projects to use `react-native-tvos` 0.62.

Reviewer 2: ```suggestion
Releases of `react-native-tvos` will be based on the public release of React Native. For this 0.62 release of `react-native` please upgrade Apple TV projects to use `react-native-tvos` 0.62.
```

> ## Other Improvements

Reviewer: as briefly mentioned in the Discord, I think that maybe it would be better to add a section (and I think the best spot would be here, before 'Other improvements') to talk about the new upgrading support repo. Something along the lines of what has been written here https://github.com/facebook/react-native-website/pull/1576

But reduced to one or 2 paragraphs and let the "heavy work" of explaining to the readme over there.

Something like:
```suggestion
## Even more upgrading support

When 0.61 was released, the community introduced a new tool called [upgrade-helper](https://react-native-community.github.io/upgrade-helper/) to better support developers in their upgrading effort. It has been a great addition and we are grateful for all the positive vibes that we felt around that project.

That said, given that upgrading can still be an hard task because of the inevitable evolution of the native platforms, we have decided to one-up the dedicated upgrading resources: we are eager to announce [Upgrade-Support](https://github.com/react-native-community/upgrade-support), a central place to request and give help when upgrading your app.

Upgrade Support is a GitHub repo hosting an issue tracker where people can post any issues they have while upgrading their projects. This should help increase the likelihood that people can find others with the same problems and work together to figure out a solution. For more information around this project, please refer to its [README](https://github.com/react-native-community/upgrade-support).

## Other Improvements
```

> - **LogBox**: We’re adding the new LogBox error and warning experience as an opt-in using unstable_enableLogBox.

Reviewer 5: Is there a LogBox docs page or README that we can link to from here, or is this something that we don't want to encourage lay users to opt in to?

Reviewer: We have been writing how to enable it in the releases https://github.com/facebook/react-native/releases

so I'd say that we could have the line here, as it's basically mentioned already.

Reviewer: so maybe something like 
```suggestion
- **LogBox**: We’re adding the new LogBox error and warning experience as an opt-in; to enable it, add `require('react-native').unstable_enableLogBox()` to your `index.js` file.
```

- **React DevTools v4**: This change includes an upgrade to the [latest React DevTools](https://reactjs.org/blog/2019/08/15/new-react-devtools.html) which offers significant performance gains, an improved navigation experience, and full support for React Hooks.

> - **Accessibility improvements**: We’ve made improvements to accessibility including adding [accessibilityValue](https://github.com/facebook/react-native/commit/7df3eea1a79f12c2dfff1976d0cef605a83232ec), missing props on [Touchables](https://github.com/facebook/react-native/commit/8c0c860e38f57e18296f689e47dfb4a54088c260), onSlidingComplete [accessibility events](https://github.com/facebook/react-native/commit/c7aa6dc8270c0eabc913fe6c617c8131e3f4b3c5), and changing the default role of Switch component from ‘button’ to ‘swtich’

Reviewer 5: I think I would link to the accessibilityValue docs here rather than the commits. (also misspelled "switch")
```suggestion
- **Accessibility improvements**: We’ve made improvements to accessibility including adding [accessibilityValue](https://reactnative.dev/docs/accessibility#accessibilityvalue-ios-android), missing props on [Touchables](https://github.com/facebook/react-native/commit/8c0c860e38f57e18296f689e47dfb4a54088c260), onSlidingComplete [accessibility events](https://github.com/facebook/react-native/commit/c7aa6dc8270c0eabc913fe6c617c8131e3f4b3c5), and changing the default role of Switch component from ‘button’ to ‘switch’.
```

> - **Accessibility improvements**: We’ve made improvements to accessibility including adding [accessibilityValue](https://github.com/facebook/react-native/commit/7df3eea1a79f12c2dfff1976d0cef605a83232ec), missing props on [Touchables](https://github.com/facebook/react-native/commit/8c0c860e38f57e18296f689e47dfb4a54088c260), onSlidingComplete [accessibility events](https://github.com/facebook/react-native/commit/c7aa6dc8270c0eabc913fe6c617c8131e3f4b3c5), and changing the default role of Switch component from ‘button’ to ‘swtich’

Reviewer 2: ```suggestion
- **Accessibility improvements**: We’ve made improvements to accessibility including adding [accessibilityValue](https://github.com/facebook/react-native/commit/7df3eea1a79f12c2dfff1976d0cef605a83232ec), missing props on [Touchables](https://github.com/facebook/react-native/commit/8c0c860e38f57e18296f689e47dfb4a54088c260), `onSlidingComplete` [accessibility events](https://github.com/facebook/react-native/commit/c7aa6dc8270c0eabc913fe6c617c8131e3f4b3c5), and changing the default role of the [Switch](https://reactnative.dev/docs/switch) component from ‘button’ to ‘swtich’
```

(Adds backticks to `onSlidingComplete`, adds ‘the’ to ‘Switch component’, and links to the Switch component doc.)

> - **Accessibility improvements**: We’ve made improvements to accessibility including adding [accessibilityValue](https://reactnative.dev/docs/accessibility#accessibilityvalue-ios-android), missing props on [Touchables](https://github.com/facebook/react-native/commit/8c0c860e38f57e18296f689e47dfb4a54088c260), onSlidingComplete [accessibility events](https://github.com/facebook/react-native/commit/c7aa6dc8270c0eabc913fe6c617c8131e3f4b3c5), and changing the default role of Switch component from ‘button’ to ‘switch’.

Author: ```suggestion
- **Accessibility improvements**: We’ve made improvements to accessibility including adding [accessibilityValue](https://reactnative.dev/docs/accessibility#accessibilityvalue-ios-android), missing props on [Touchables](https://github.com/facebook/react-native/commit/8c0c860e38f57e18296f689e47dfb4a54088c260), `onSlidingComplete` [accessibility events](https://github.com/facebook/react-native/commit/c7aa6dc8270c0eabc913fe6c617c8131e3f4b3c5), and changing the default role of Switch component from ‘button’ to ‘switch’.
```

## Breaking Changes

> - **Remove PropTypes**: We're removing propTypes from core components in order to release the app size impact of React Native core and to favor static type systems which check at compile time instead of runtime.

Reviewer 6: In order to _reduce_ the app size impact?

Author: Fixed ty

> - **Remove PropTypes**: We're removing propTypes from core components in order to release the app size impact of React Native core and to favor static type systems which check at compile time instead of runtime.

Reviewer 2: ```suggestion
- **Remove PropTypes**: We're removing `propTypes` from core components in order to release the app size impact of React Native core and to favor static type systems which check at compile time instead of runtime.
```

- **Remove accessibilityStates**: We’ve [removed](https://github.com/facebook/react-native/commit/7b35f427fd66cb0f36921b992095fe5b3c14d8b9) the deprecated `accessibilityStates` property in favor of the new `accessibilityState` prop which is a more semantically rich way for components to describe information about their state to accessibility services.

> - **TextInput changes**: We removed `onTextInput` [from TextInput](https://github.com/facebook/react-native/commit/3f7e0a2c9601fc186f25bfd794cd0008ac3983ab) as it’s uncommon, not W3c compliant, and difficult to implement in Fabric. We also removed the undocumented `inputView` prop, and `selectionState`.

Reviewer 2: ```suggestion
- **TextInput changes**: We removed `onTextInput` [from TextInput](https://github.com/facebook/react-native/commit/3f7e0a2c9601fc186f25bfd794cd0008ac3983ab) as it’s uncommon, not W3C compliant, and difficult to implement in [Fabric](https://github.com/react-native-community/discussions-and-proposals/issues/4). We also removed the undocumented `inputView` prop, and `selectionState`.
```

## Deprecations

- `AccessibilityInfo.fetch` was already deprecated, but in this release we [added a warning](https://github.com/facebook/react-native/commit/523ab8333800afbfb169c6fd70ab6611fe07cc2a).
- Setting `useNativeDriver` is [now required](https://github.com/facebook/react-native/commit/5876052615f4858ed5fc32fa3da9b64695974238) to support switching the default in the future.

> - Refs on an `Animated` component are now the internal component and `getNode` has [been deprecated](https://github.com/facebook/react-native/commit/66e72bb4e00aafbcb9f450ed5db261d98f99f82a).

Reviewer 2: ```suggestion
- The `ref` of an `Animated` component is now the internal component and `getNode` has [been deprecated](https://github.com/facebook/react-native/commit/66e72bb4e00aafbcb9f450ed5db261d98f99f82a).
```

## Thanks

Thanks to the hundreds of contributors that helped make 0.62 possible!

> To see all the updates, take a look at the [0.62 changelog](https://github.com/react-native-community/releases/blob/master/CHANGELOG.md).

Reviewer 2: ```suggestion
To see all the updates, take a look at the [0.62 changelog](https://github.com/react-native-community/releases/blob/master/CHANGELOG.md#0620).
```

---
title: Announcing React Native 0.62 with Flipper
author: Rick Hanlon
authorTitle: React Native Core at Facebook
authorURL: https://twitter.com/rickhanlonii
authorImageURL: https://avatars3.githubusercontent.com/u/2440089?s=460&v=4
authorTwitter: rickhanlonii
category: announcements
---

Today we’re releasing React Native version 0.62 which includes support for Flipper by default.

This release comes in the midst of a global pandemic. We’re releasing this version today to respect the work of hundreds of contributors who made this release possible and to prevent the release from falling too far behind master. Please be mindful of the reduced capacity of contributors to help with issues and prepare to delay upgrading if necessary.

## Flipper by default

[Flipper](https://fbflipper.com/) is a developer tool for debugging mobile apps. It’s popular in the Android and iOS communities, and in this release we’ve enabled support by default for new and existing React Native apps.

Flipper provides the following features out of the box:

- Network Inspector
- Layout Inspector
- Crash Reporter
- Device logs

> There is also a marketplace of custom Flipper plugins you can install:

Reviewer 3: ```suggestion
Flipper provides the following features out of the box:

- Reloading the app and triggering the Dev Menu.
- React DevTools
- Metro Logs
- Device Logs
- Crash Reporter
- Database Inspector
- Preference Inspector

Since Flipper is an extensible platform, there is a built-in marketplace (connected to NPM) which could feature your plugins!
```

I removed the the Network Inspector, as, IIRC, the RN network stack doesn't use the underlying libraries that the Network Inspector uses, so afaik it doesn't show anything useful normally? Or does it work in your experience?

I removed the Layout Inspector as well, it does work, but the value to RN users isn't really clear to me. If you think it is a valuable feature, feel free to put it back. 

cc @passy

Author: That makes sense -- I'll add the layout inspector back as this is super useful for debugging layout issues in React Native 👍

What would need to be done to get the Network plugin working?

Reviewer 3: Afaik, but I'm on very unfamiliar territory now, it intercepts the `okhttp3` library on Android, and the native libraries on iOS. But I am not sure that RN Android uses that library at all for networking (sorry, I don't have a RN non-FB app ready at hand that does send network requests, I can gobble something together tomorrow if you want me to actually test that)

Reviewer 7: I thought RN did use okhttp?

Reviewer 7: It does. :) If this doesn't work it's a bug, but I haven't heard that from anyone who's used it so far.

Reviewer 3: Just tried it, and networking plugin works on both RN and android, so we should definitely mention it :)

Author: Nice, thanks! I also consolidated and added descriptions to each to fill out the content and provide context where it might not be obvious.

## Other Improvements

- **LogBox**: We’re adding the new LogBox error and warning experience as an opt-in; to enable it, add `require('react-native').unstable_enableLogBox()` to your `index.js` file.
- **React DevTools v4**: This change includes an upgrade to the [latest React DevTools](https://reactjs.org/blog/2019/08/15/new-react-devtools.html) which offers significant performance gains, an improved navigation experience, and full support for React Hooks.
- **Accessibility improvements**: We’ve made improvements to accessibility including adding [accessibilityValue](https://reactnative.dev/docs/accessibility#accessibilityvalue-ios-android), missing props on [Touchables](https://github.com/facebook/react-native/commit/8c0c860e38f57e18296f689e47dfb4a54088c260), `onSlidingComplete` [accessibility events](https://github.com/facebook/react-native/commit/c7aa6dc8270c0eabc913fe6c617c8131e3f4b3c5), and changing the default role of Switch component from ‘button’ to ‘switch’.

## Breaking Changes

> - **Remove PropTypes**: We're removing `propTypes` from core components in order to release the app size impact of React Native core and to favor static type systems which check at compile time instead of runtime.

Author: ```suggestion
- **Remove PropTypes**: We're removing `propTypes` from core components in order to reduce the app size impact of React Native core and to favor static type systems which check at compile time instead of runtime.
```
